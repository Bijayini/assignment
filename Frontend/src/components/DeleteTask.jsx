import React, {Component, Fragment} from 'react';
import PropTypes from "prop-types";

import Button from "../ui-components/Button";
import Modal from "../ui-components/Modal";

import labels from "../labels/Task.labels";

class DeleteTask extends Component{
    constructor(){
        super();
        this.state = {
            isOverlayOpen:false
        }
    }

    toggleOverlay=()=>{
        this.setState({
            isOverlayOpen:!this.state.isOverlayOpen
        })
    };

    handelDelete = () =>{
        const {handelClick, id } = this.props;
        handelClick(id);
        this.toggleOverlay();
    };


    render(){
        const {isOverlayOpen} = this.state;
        const{id } = this.props;
        const buttons = [
            {
                label:labels.cancel,
                type: 'secondary',
                callback:this.toggleOverlay,
            },
            {
                label: labels.continue,
                type: 'primary',
                callback:()=>{this.handelDelete(id)},
            },
        ];
        return(<Fragment>
                <Button variant="primary" size="small"  onClick={this.toggleOverlay}>{labels.delete}</Button>
                {isOverlayOpen && <Modal toggleOverlay={this.toggleOverlay} buttons={buttons}>
                    <p>{labels.are_you_sure}</p>
                </Modal>}
            </Fragment>

           );
    }
}

DeleteTask.propTypes = {
    handelClick: PropTypes.func.isRequired,
};

export default DeleteTask;