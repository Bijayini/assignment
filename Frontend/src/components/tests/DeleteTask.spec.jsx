import React from 'react';
import { shallow } from 'enzyme';
import DeleteTask from '../DeleteTask';


describe('DeleteTask', () => {
  let component;
  let props;

  beforeEach(()=>{
    props = {
      id:34,
      handelClick:jest.fn()
    };
    component = shallow(<DeleteTask {...props}/>);
  });

  it('should match with snapshot', () => {
    expect(component.debug()).toMatchSnapshot();
  });

  it('should call handelClick on handelDelete', ()=>{
    component.instance().handelDelete();

    expect(props.handelClick).toHaveBeenCalledWith(34);
  })
});
