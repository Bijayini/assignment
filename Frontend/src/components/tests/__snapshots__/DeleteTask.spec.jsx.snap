// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteTask should match with snapshot 1`] = `
"<Fragment>
  <Button variant=\\"primary\\" size=\\"small\\" onClick={[Function]} type=\\"button\\" classList=\\"\\">
    Delete
  </Button>
</Fragment>"
`;
